applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -84,6 +84,11 @@ class TrilinosBlockBuilderAndSolver
 public:
     ///@name Type Definitions
     ///@{
+
+    /// Definition of the flags
+    KRATOS_DEFINE_LOCAL_FLAG( SILENT_WARNINGS );
```

Comment:
Considering that we are in general promoting the use of parameter-based constructors and that the local flags have been always problematic (at least IMO), specially in presence of template arguments, I'd substitute this by a member variable that takes the value provided in the input json.

This of course would preclude setting the value if the non-parameters constructor is used. However, this is already the current behavior so I don't see any problem on it.

Reply:
> I've some concerns
> 
>     * The issue with the local flag.

It is the smae used in the serial B&S. It is consistent.

>     * I'd minimize the changes in such crucial and delicate class. This means to remove all the aesthetics in `ApplyDirichletConditions` method to really make clear which are the required changes. We can merge these in a future PR.

These are minimal changes, https://github.com/KratosMultiphysics/Kratos/pull/10887 these are the transition changes.

>     * The test results are changed. I guess this is required as you're now checking not only the "standard" values but also the Dirichlet substituted ones. Noting in this regard that as far as I see, the old values remain the same. Can you please confirm?

Consistent with serial B&S (we changed recently!)

>     * Besides, the PR also introduces extend current tests to check the scaling. I'd add these as new standalone tests (even though this requires copy&pasting some code) to ease an eventual need of debugging.

The method to scale is already tested and was added a long time ago.

Reply:
Hello @rubenzorrilla

Reply:
> > I've some concerns
> > ```
> > * The issue with the local flag.
> > ```

OK. It makes sense to me to make it consistent with the serial one in a first step to then move towards its removal.

> 
> It is the smae used in the serial B&S. It is consistent.
> 
> > ```
> > * I'd minimize the changes in such crucial and delicate class. This means to remove all the aesthetics in `ApplyDirichletConditions` method to really make clear which are the required changes. We can merge these in a future PR.
> > ```
> 
> These are minimal changes, #10887 these are the transition changes.
> 

I insist on this. Aesthetics must be absolutely avoided when changing such crucial and delicate classes. If they're that minimal it should be straightforward to remove them.

> > ```
> > * The test results are changed. I guess this is required as you're now checking not only the "standard" values but also the Dirichlet substituted ones. Noting in this regard that as far as I see, the old values remain the same. Can you please confirm?
> > ```
> 
> Consistent with serial B&S (we changed recently!)
> 

Sorry but I can't understand your answer.

> > ```
> > * Besides, the PR also introduces extend current tests to check the scaling. I'd add these as new standalone tests (even though this requires copy&pasting some code) to ease an eventual need of debugging.
> > ```
> 
> The method to scale is already tested and was added a long time ago.

Then I can't understand the changes in the tests.

Reply:
> I insist on this. Aesthetics must be absolutely avoided when changing such crucial and delicate classes. If they're that minimal it should be straightforward to remove them.

I will try to move those changes to an independent PR.

> > > * The test results are changed. I guess this is required as you're now checking not only the "standard" values but also the Dirichlet substituted ones. Noting in this regard that as far as I see, the old values remain the same. Can you please confirm?
> > > ```
> > 
> > 
> > Consistent with serial B&S (we changed recently!)
> 
> Sorry but I can't understand your answer.

The Dirichlet values are affected because this is the pourpose of the PR, unify the behaviour and therefore Dirichlet is modified in the same manner as in serial. You can see the results are the same as in serial test: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp

> > > ```
> > > * Besides, the PR also introduces extend current tests to check the scaling. I'd add these as new standalone tests (even though this requires copy&pasting some code) to ease an eventual need of debugging.
> > > ```
> > 
> > 
> > The method to scale is already tested and was added a long time ago.
> 
> Then I can't understand the changes in the tests.

Tests are exactly the same as in serial, see: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp

In addition, the scaling is tested standalone manner: https://github.com/KratosMultiphysics/Kratos/blob/eb3f0c7e144085c522a2a8f2bbc3c25e8ffa31d1/applications/TrilinosApplication/tests/cpp_tests/test_trilinos_space.cpp#L559